test/ui/test_spec.js

```diff
@@ -96,29 +94,42 @@ describe('UI tests', function() {
     });
 
     it('appears correctly before any actions have been taken', function(done) {
-      return screenshotAndCompare('body.png', 'body', done);
+      screenshotAndCompare('body.png', 'body')
+        .then(function() {
+          return done();
+        });
     });
 
     it('opens help menu correctly when its button is clicked', function(done) {
       return driver.findElement(By.id('helpButton'))
         .then(function(button) {
-          button.click();
+          return button.click();
         })
-        .then(screenshotAndCompare('bodyWithHelp.png', 'bodyWithHelp', done));
+        .then(function() {
+          return screenshotAndCompare('bodyWithHelp.png', 'bodyWithHelp');
+        })
+        .then(function() {
+          return done();
+        });
     });
 
     it('appears correctly after closing help menu by clicking the body element', function(done) {
       return driver.findElement(By.tagName('body'))
         .then(function(button) {
-          button.click();
+          return button.click();
```

Comment:
If you use the new syntax, I think you could write these lines like this:
return driver.findElement(By.tagName('body'))
  .then(button => button.click())
  .then(() => screenshotAndCompare('body.png', 'body'))
  .then(done);
Just a style issue - your choice.
The same comment applies for the other changes you made below, where there are now a bunch of one-line functions that could be rewritten as actual one-lines.